pkg/scalers/couchdb_scaler.go

```
@@ -0,0 +1,247 @@
+	default:
+		meta.connectionString = ""
+		meta.host, err = GetFromAuthOrMeta(config, "host")
+		if err != nil {
+			return nil, "", err
+		}
+
+		meta.port, err = GetFromAuthOrMeta(config, "port")
+		if err != nil {
+			return nil, "", err
+		}
+
+		meta.username, err = GetFromAuthOrMeta(config, "username")
+		if err != nil {
+			return nil, "", err
+		}
+
+		if config.AuthParams["password"] != "" {
+			meta.password = config.AuthParams["password"]
+		} else if config.TriggerMetadata["passwordFromEnv"] != "" {
+			meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
+		}
+		if len(meta.password) == 0 {
+			return nil, "", fmt.Errorf("no password given")
+		}
+	}
+
+	if meta.connectionString != "" {
+		connStr = meta.connectionString
+	} else {
+		// Build connection str
+		addr := fmt.Sprintf("%s:%s", meta.host, meta.port)
+		// auth := fmt.Sprintf("%s:%s", meta.username, meta.password)
+		connStr = "http://" + addr
+	}
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("couchdb-%s", val))
+	}
+	meta.scalerIndex = config.ScalerIndex
+	return &meta, connStr, nil
+}
+
+func NewCouchDBScaler(ctx context.Context, config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+
+	meta, connStr, err := parseCouchDBMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parsing couchDB metadata, because of %v", err)
+	}
+
+	client, err := kivik.New("couch", connStr)
+	if err != nil {
+		fmt.Println("Couldn't connect to couchdb: ", err)
+		return nil, fmt.Errorf("%v", err)
+	}
+
+	err = client.Authenticate(context.TODO(), couchdb.BasicAuth("admin", meta.password))
+	if err != nil {
+		fmt.Println("Couldn't authenticate: ", err)
+	} else {
+		fmt.Println("Authenticated..")
+	}
```

If there is an error, shouldn't we return the error and fail the scaler?
Apart from that, please use the logger instead of printing the lines